File: apps/pwa/src/server/base/login_with_2fa.ts
import { request, Method } from '..';
import type { LoginResponse } from './login';

interface RequestBody {
  username: string;
  password: string;
  twoFAToken: string;
  deviceName?: string;
}

function loginWith2FA(data: RequestBody) {
  return request<LoginResponse>({
    path: '/base/login_with_2fa',
    method: Method.POST,
    body: {
      username: data.username,
      password: data.password,
      twoFAToken: data.twoFAToken,
      deviceName: data.deviceName,
    },
  });
}

export default loginWith2FA;

File: apps/flutter/lib/server/base/login_with_2fa.dart
import '../request.dart';

Future<String> loginWith2FA({
  required String username,
  required String password,
  required String twoFAToken,
}) async {
  final token = await httpPost(
    path: "/base/login_with_2fa",
    data: {
      "username": username,
      "password": password,
      "twoFAToken": twoFAToken,
    },
  );
  return token;
}
